src/org/jetbrains/plugins/scala/codeInsight/intention/booleans/ReplaceEqualsOrEqualityInMethodCallExprIntention.scala

```diff
@@ -55,12 +62,100 @@ class ReplaceEqualsOrEqualityInMethodCallExprIntention extends PsiElementBaseInt
     val oper = methodCallExpr.getInvokedExpr.asInstanceOf[ScReferenceExpression].nameId.getText
 
     expr.append(methodCallExpr.getInvokedExpr.asInstanceOf[ScReferenceExpression].
-            qualifier.get.getText).append(".").append(replaceOper(oper)).append(methodCallExpr.args.getText)
+      qualifier.get.getText).append(".").append(replaceOper(oper)).append(methodCallExpr.args.getText)
 
     val newMethodCallExpr = ScalaPsiElementFactory.createExpressionFromText(expr.toString(), element.getManager)
 
     val size = newMethodCallExpr.asInstanceOf[ScMethodCall].getInvokedExpr.asInstanceOf[ScReferenceExpression].nameId.
+      getTextRange.getStartOffset - newMethodCallExpr.getTextRange.getStartOffset
+
+    inWriteAction {
+      methodCallExpr.replace(newMethodCallExpr)
+      editor.getCaretModel.moveToOffset(start + size)
+      PsiDocumentManager.getInstance(project).commitDocument(editor.getDocument)
+    }
+  }
+
+  def invoke_method_one(project: Project, editor: Editor, element: PsiElement) {
+    val methodCallExpr: ScMethodCall = PsiTreeUtil.getParentOfType(element, classOf[ScMethodCall], false)
+    if (methodCallExpr == null || !methodCallExpr.isValid) return
+
+    val start = methodCallExpr.getTextRange.getStartOffset
+
+    val replaceOper = Map("equals" -> "==", "==" -> "equals")
+
+    val args = new StringBuilder().append(methodCallExpr.args.getText)
+
+    val oper = methodCallExpr.getInvokedExpr.asInstanceOf[ScReferenceExpression].nameId.getText
+    val desiredOper: String = replaceOper(oper)
+
+    val expr = new StringBuilder()
+      .append(methodCallExpr.getInvokedExpr.asInstanceOf[ScReferenceExpression].qualifier.get.getText)
+
+    if (desiredOper == "==") {
+      expr.append(" ")
```

Comment:
I think we should remove or refactor usages of `IntentionUtils.analyzeMethodCallArgs`. It's too java-style and has a misleading name. So in that regard I found `invoke_method_two` better.